src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -2548,7 +2574,12 @@ class DropStatementSegment(BaseSegment):
             "USER",
         ),
         Ref("IfExistsGrammar", optional=True),
-        Ref("TableReferenceSegment"),
+        OneOf(
+            # TABLE/VIEW
+            Ref("TableReferenceSegment"),
+            # User
+            Ref("ObjectReferenceSegment"),
+        ),
```

Comment:
`TableReferenceSegment` inherits `ObjectReferenceSegment` so the parser would always just match `TableReferenceSegment` here. That means you still get the user being interpreted as a table.

It'll be a lot cleaner to handle this when you split this segment up though 😄